### ISSUE TITLE
Android Publishing Gets Stuck on Archiving

### ISSUE THREAD

**Issue by LeoJHarris** [dotnet/android#9540]
### Description

I've waited nearly an hour on an android project to try publish a release build and its taken so long I gave up on it so posting my concern here. As this is my first attempt on a MAUI release build I cannot comment on previous versions.

Stuck on here:

![Image](https://github.com/user-attachments/assets/8cd42f18-bf7a-45c8-aee3-28379f76768a)

### Steps to Reproduce

1. Right click project
2. Select Publish...
3. Archive...

### Version with bug

9.0.10 SR1

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Did you find any workaround?

NO

### Relevant log output

```shell

```

**Comment by jpobst** [comment]
Any issue(s) with the VS Archiving experience would be owned by the VS integration team who does not use GitHub for issue tracking.  

Please use the [tools available inside of VS](https://developercommunity.visualstudio.com/VisualStudio/report) to report this issue.  This allows the issue to be routed to the correct team, and provides them the correct logs they need to diagnose it.

---

As for the command line error:

```
error XA1018: Specified AndroidManifest file does not exist: C:\xxxx\Platforms\Android\Release\AndroidManifest.xml.
```

@jonathanpeppers @dellis1972 Would this be expected if custom configurations are being used? Or do we possibly have a bug here?

**Comment by LeoJHarris** [comment]
@jonathanpeppers We do something like the following to get the full set of properties depending on the Constants defined.

```csproj
<PropertyGroup Condition="'$(Configuration)'=='ReleaseFoo'">
	<DefineConstants>FOO;RELEASE;</DefineConstants>
</PropertyGroup>

<Choose>
	<When Condition="$(DefineConstants.Contains('FOO'))">
                <PropertyGroup Condition="'$(TargetFramework)' == 'net8.0-android'">
                	<AndroidManifest>Platforms\Android\FOO\AndroidManifest.xml</AndroidManifest>
                </PropertyGroup>
        </When>
<Otherwise>
```

Also, I have the MSBuild_Logs folder, is there a particular .binlog file you would like or the entire folder? To avoid exposing sensitive information should I share this with you directly?

**Comment by jonathanpeppers** [comment]
You can share with my email on my Github repo if you like. Can you just use `Configuration=Release` for everything? And then invent a new property like `-p:Foo=true` to make changes for a special build?

**Comment by LeoJHarris** [comment]
@jonathanpeppers Maybe but I don't see how that's a whole lot different to specifying as a configuration i.e. `Configuration=Foo` vs `-p:Foo=true` and correct if I am wrong but I lose the benefit of the `DefineConstants` when I can toggle between the configuration selected.

**Comment by LeoJHarris** [comment]
@jonathanpeppers correction I might have confused myself here, what your saying here is I can just use the custom property `-p` for selecting the correct signing identities and the manifests/info.plist files?

**Comment by jonathanpeppers** [comment]
Yes, using any property other than $(Configuration) would work. Otherwise you would need to review a .binlog from a build to discover the set of properties for a Debug-like configuration vs a Release-like one. No one really knows the full list as noted in the dotnet/sdk issue above.

**Comment by LeoJHarris** [comment]
As of now I can do the publishing via the cmd line without any issues by specifying the config, as I'm a little snowed under other things Ill try revisit this again later and if this becomes an issue.
